### PR TITLE
Documentation added for extended ADC config functionality

### PR DESCRIPTION
**Pull request by mpcender**
### Linked Jira Issue:
https://ser401-2022-group39.atlassian.net/browse/MALS-190

### Description of Changes:
Added document describing the functionality added to the DxCore for ADC.

### Acceptance Criteria:
- [x] AC1: A single document describing all added ADC features has been created
- [x] AC2: Document has been placed in the DxCore repository

### CODE REVIEW

File: megaavr/extras/Ref_ADC_config.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# ADC API Design
+The ADC API design in the DxCore library introduces new data structures and 
+functions for configuring and reading analog values from the ADC of a 
+microcontroller or similar device. The new data structure, ADCConfig, allows for 
+easy configuration of various ADC parameters such as conversion mode, resolution, 
+prescaler, sample delay, sample length, and sample number. Additionally, new 
+functions such as analogRead(), analogReadConfig(), getAnalogReadConfig(), 
+analogReadSampleDelay(), analogReadEnableSingleEnded(), analogReadEnableDifferential(), 
+analogReadSampleNum(), and analogReadPrescaler() are introduced for reading and
```
Comment (jnbrauer) [inline megaavr/extras/Ref_ADC_config.md:9]: Let's put these function names, and the struct name above, in code tags.

File: megaavr/extras/Ref_ADC_config.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+Structure for containing all of the ADC configuration parameters:
+```c
+typedef struct ADCConfig_struct {
+    uint8_t convMode;    // ADC conversion mode (single-ended/differential)
+	uint8_t muxNeg;      // Negative pin for differential ADC conversion
+	uint8_t resolution;  // ADC bit resolution
+	uint8_t prescaler;   // ADC clock prescaler value
+	uint8_t sampleNum;   // Number of ADC samples to accumulate
+	uint8_t sampleDelay; // Delay between ADC samples in accumulation mode
+	uint8_t sampleLen;   // ADC sampling time in ADC clock cycles
```
Comment (jnbrauer) [inline megaavr/extras/Ref_ADC_config.md:23]:
```suggestion
    uint8_t muxNeg;      // Negative pin for differential ADC conversion
    uint8_t resolution;  // ADC bit resolution
    uint8_t prescaler;   // ADC clock prescaler value
    uint8_t sampleNum;   // Number of ADC samples to accumulate
    uint8_t sampleDelay; // Delay between ADC samples in accumulation mode
    uint8_t sampleLen;   // ADC sampling time in ADC clock cycles
```

File: megaavr/extras/Ref_ADC_config.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+Valid values for `samples`:
+- 0x0: no accumulation (`ADC_SAMPNUM_NONE_gc`)
+- 0x1: 2 samples (`ADC_SAMPNUM_ACC2_gc`)
+- 0x2: 4 samples (`ADC_SAMPNUM_ACC4_gc`)
+- 0x3: 8 samples (`ADC_SAMPNUM_ACC8_gc`)
+- 0x4: 16 samples (`ADC_SAMPNUM_ACC16_gc`)
+- 0x5: 32 samples (`ADC_SAMPNUM_ACC32_gc`)
+- 0x6: 64 samples (`ADC_SAMPNUM_ACC64_gc`)
+- 0x7: 128 samples (`ADC_SAMPNUM_ACC128_gc`)
```
Comment (jnbrauer) [inline megaavr/extras/Ref_ADC_config.md:102]: I believe this is incorrect, as this is implemented now you pass in the actual desired number of samples. Also, let's avoid referencing the internal constants (`ADC_SAMPNUM_NONE_gc` etc.), most users probably won't know where those are coming from.

File: megaavr/extras/Ref_ADC_config.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+- 0x4: clock divided by 16 (`ADC_PRESC_DIV16_gc`)
+- 0x5: clock divided by 20 (`ADC_PRESC_DIV20_gc`)
+- 0x6: clock divided by 24 (`ADC_PRESC_DIV24_gc`)
+- 0x7: clock divided by 28 (`ADC_PRESC_DIV28_gc`)
+- 0x8: clock divided by 32 (`ADC_PRESC_DIV32_gc`)
+- 0x9: clock divided by 48 (`ADC_PRESC_DIV48_gc`)
+- 0xA: clock divided by 64 (`ADC_PRESC_DIV64_gc`)
+- 0xB: clock divided by 96 (`ADC_PRESC_DIV96_gc`)
+- 0xC: clock divided by 128 (`ADC_PRESC_DIV128_gc`)
+- 0xD: clock divided by 256 (`ADC_PRESC_DIV256_gc`)
```
Comment (jnbrauer) [inline megaavr/extras/Ref_ADC_config.md:125]: Same comments as above.

**Review by jnbrauer — CHANGES_REQUESTED** [review]

**Comment by mpcender** [comment]
Thanks for the notes! Good eye on the function input params. I hadn't realized those had changed from the original API design. Should all be updated

**Review by jnbrauer — APPROVED** [review]
Looks good!

**Pull request closed without merging**